### PR TITLE
SI-6756 A reify test case and and an improvement

### PR DESCRIPTION
**Pull request by retronym**
Review by @densh or @xeno-by

### CODE REVIEW

File: src/reflect/scala/reflect/internal/StdNames.scala [hunk 1]
```diff
@@ -598,6 +598,7 @@ trait StdNames {
     val Annotation: NameType           = "Annotation"
     val Any: NameType                  = "Any"
     val AnyVal: NameType               = "AnyVal"
+    val AppliedTypeTree: NameType      = "AppliedTypeTree"
```
Comment (xeno-by) [inline src/reflect/scala/reflect/internal/StdNames.scala]: What's this for?

File: src/compiler/scala/reflect/reify/codegen/GenTypes.scala [hunk 2]
```diff
@@ -55,7 +55,12 @@ trait GenTypes {
       case tpe @ ConstantType(value) =>
         mirrorBuildCall(nme.ConstantType, reifyProduct(value))
       case tpe @ TypeRef(pre, sym, args) =>
-        reifyBuildCall(nme.TypeRef, pre, sym, args)
+        if (pre != NoPrefix && !sym.isLocatable) {
+          val pre1 = reify(pre)
+          val sym1 = Apply(Select(pre1, nme.member), reifyName(sym.name) :: Nil)
+          mirrorBuildCall(nme.TypeRef, pre1, sym1, reify(args))
```
Comment (xeno-by) [inline src/compiler/scala/reflect/reify/codegen/GenTypes.scala:61]: This is good, LGTM.

File: src/reflect/scala/reflect/internal/StdNames.scala (hunk 1 above)
Comment (retronym) [inline src/reflect/scala/reflect/internal/StdNames.scala]: Ooops, this is detritus from my first attempt, I'll force push a removal of this line.

**Comment by retronym** [comment]
I can't figure out how to get this working without breaking other tests (`t8104|t6992|reify_newimpl_19`), so I'm going to close it for now.

**Pull request closed without merging**